packages/cells/src/model.ts

```diff
@@ -153,6 +153,11 @@ export interface ICodeCellModel extends ICellModel {
    */
   executionCount: nbformat.ExecutionCount;
 
+  /**
+   * The code cell's state.
+   */
+  executionState: 'running' | 'idle';
```

Comment:
Thanks a lot @krassowski for proposing this.

Should there be additional state:
- `pending-input` state
- `queued` state (sent for execution but not running yet)

Reply:
In the future I would love to have `queued` state. I am not sure about `pending-input` but maybe. However, I do not want to hold this PR on reaching a consensus on additional future values we might include; I think it is fair to say that we may want to include more states and hence it should not be a boolean but an enumeration of possible values.